### ISSUE TITLE
Allow recreating a model's migration

### ISSUE THREAD

**Issue by xjunior** [rails/rails#39238]
### Steps to reproduce
1) Create a model (run migrations and everything)
2) Rename the model to something else
3) Create another model with the same original name of the other model

### Expected behavior
It would allow me to do that

### Actual behavior
You get the error: `Multiple migrations have the name <Original name>.`

### System configuration
**Rails version**: Rails 5.1.6.2

**Ruby version**: 2.5.0

**Comment by rafaelfranca** [comment]
That you for the issue. I understand your case but since you renamed the model but didn't rename the migration this error will happen. Maybe we could disable the conflict detection for migration names but that is to avoid the mistake of having two migrations to the same table overriding each other.

I think it is fair to expect that when a model is renamed the migration is also renamed.

**Comment by xjunior** [comment]
@rafaelfranca I renamed the model, and didn't rename the migration because it was a concept that just changed the name later. We didn't change the migration because it was already ran in production, but there's a migration to change the table name. I understand the protection to help you not add duplicate migrations, but if we want this extra-concerned behavior we might want to look for other migrations changing the table name?

**Comment by xjunior** [comment]
In our situation this is what happened:

1) We introduced a model called `CareerChange`;
2) We later renamed it to `CareerChangeRequest`;
3) Now, we're introducing a concept which is the result of a successful request, so a `CareerChange`.

I don't see that as an invalid scenario, so I rather not have this protection to not add a migration with the same name.

**Comment by sandip-mane** [comment]
I have faced this issue in the other use case. 
We dropped all unused models before going to production, 1-2 months later we came up with a situation where we wanted to use the same name.

For this we had two options and I feel both of them are workarounds
1. Rename old migration which created the table. 
2. Manually create new migration

@rafaelfranca I will raise a PR for this if this case makes sense! Maybe we can check if the table exists in the schema?
